api/app/src/models/medical/patient.ts

```diff
@@ -54,13 +54,13 @@ export abstract class PatientExternalDataEntry {}
 export type PatientExternalData = Partial<Record<MedicalDataSource, PatientExternalDataEntry>>;
 
 export type PatientData = {
-  firstName: string;
-  lastName: string;
+  firstName: string[];
+  lastName: string[];
```

Comment:
Mainly so we don't have to remember to do joins/splits when working with names in the API, this seems like it would be more error prone.

In any case, we will need to revisit all of this soon (perhaps as part of the FHIR work), as there are several limitations as-is:

- we don't know what the "use" of name/address/contact is in any of the arrays (ie current, home, old, etc)
- we don't support multiple entries on the dashboard

This is mainly to be an improvement over what we currently have for customers currently onboarding.

Reply:
Taken to [Slack](https://metriport.slack.com/archives/C04CXLSAF7V/p1683221629071449?thread_ts=1683178646.354949&cid=C04CXLSAF7V)

Reply:
Gave this some thought, moving back to `string` and splitting/joining as suggested